scripts/api_dump_generator.py

```diff
@@ -1320,6 +1379,50 @@
     return false;
 }}
 
+std::ostream& dump_json_VkPhysicalDeviceMemoryProperties(const VkPhysicalDeviceMemoryProperties& object, const ApiDumpSettings& settings, int indents)
+{{
+    settings.stream() << settings.indentation(indents) << "[\\n";
+
+    bool needMemberComma = false;
+    if (needMemberComma) settings.stream() << ",\\n";
+    dump_json_value<const uint32_t>(object.memoryTypeCount, settings, "uint32_t", "memoryTypeCount", indents + 1, dump_json_uint32_t);
+    needMemberComma = true;
+    if (needMemberComma) settings.stream() << ",\\n";
+    dump_json_array<const VkMemoryType>(object.memoryTypes, object.memoryTypeCount, settings, "VkMemoryType[VK_MAX_MEMORY_TYPES]", "VkMemoryType", "memoryTypes", indents + 1, dump_json_VkMemoryType);
+    needMemberComma = true;
+    if (needMemberComma) settings.stream() << ",\\n";
+    dump_json_value<const uint32_t>(object.memoryHeapCount, settings, "uint32_t", "memoryHeapCount", indents + 1, dump_json_uint32_t);
+    needMemberComma = true;
+    if (needMemberComma) settings.stream() << ",\\n";
+    dump_json_array<const VkMemoryHeap>(object.memoryHeaps, object.memoryHeapCount, settings, "VkMemoryHeap[VK_MAX_MEMORY_HEAPS]", "VkMemoryHeap", "memoryHeaps", indents + 1, dump_json_VkMemoryHeap);
+    needMemberComma = true;
+    settings.stream() << "\\n" << settings.indentation(indents) << "]";
+    return settings.stream();
+}}
+
+std::ostream& dump_json_VkPhysicalDeviceGroupProperties(const VkPhysicalDeviceGroupProperties& object, const ApiDumpSettings& settings, int indents)
+{{
+    settings.stream() << settings.indentation(indents) << "[\\n";
+
+    bool needMemberComma = false;
```

Comment:
Generated code needs to use the needMemberComma variable to keep track of where commas are to be inserted between members of structs. (And not before the first member and not after the last member). Since this function is not really generated code and we know where commas are needed, I think that the needMemberComma variable can be eliminated here. Doing so would make this code much simpler, straightforward, and easier to read.

This should also be done for the previous function dump_json_VkPhysicalDeviceMemoryProperties